xchange-stream-poloniex2/src/main/java/info/bitrich/xchangestream/poloniex2/PoloniexStreamingService.java

```diff
@@ -124,21 +150,35 @@ protected String getChannelNameFromMessage(JsonNode message) {
 
   @Override
   public String getSubscribeMessage(String channelName, Object... args) throws IOException {
-    PoloniexWebSocketSubscriptionMessage subscribeMessage =
-        new PoloniexWebSocketSubscriptionMessage("subscribe", channelName);
+    Object subscribeMessage;
+    if (ACCOUNT_NOTIFICATIONS_CHANNEL.equals(channelName)) {
+      subscribeMessage =
+          getAccountNotificationsSubscription(
+              (String) args[0], (String) args[1], (SynchronizedValueFactory<Long>) args[2]);
```

Comment:
For these args casts, probably makes sense to check the array is the size you expect and each arg is of the type you expect and log when one of those checks is incorrect. This will make debugging incorrect params much simpler, `CoinbaseProMarketDataService.getOrderbook()` as an example.

Reply:
added validation